runtime/parachains/src/inclusion_inherent.rs

```diff
@@ -118,3 +123,17 @@ decl_module! {
 		}
 	}
 }
+
+impl<T: Trait> ProvideInherent for Module<T> {
+	type Call = Call<T>;
+	type Error = MakeFatalError<()>;
+	const INHERENT_IDENTIFIER: InherentIdentifier = inclusion_inherent::INHERENT_IDENTIFIER;
+
+	fn create_inherent(data: &InherentData) -> Option<Self::Call> {
+		data.get_data(&Self::INHERENT_IDENTIFIER)
+			.expect("inclusion inherent data failed to decode")
+			.map(|(signed_bitfields, backed_candidates)| {
+				Call::inclusion(signed_bitfields, backed_candidates)
```

Comment:
I've been talking with @bkchr about a bug related to a race between the off-chain backing work and session changes (which can happen between blocks). The short story is that this inherent data can actually be _invalid_ for the inherent that we are building.

The quick fix is to do a sanity check at this point, where we see if `Self::inclusion(signed_bitfields, backed_candidates)` returns `Ok(())`. If not, we return a `Call::inclusion` with `Vec::new` for both backed candidates and bitfields.

This is because we need to avoid creating an invalid block at all costs.

Reply:
Implemented in 6b88341.